### ISSUE TITLE
Escaping variable substitution not working with "\" on ONBUILD

### ISSUE THREAD

**Issue by marcellodesales** [hadolint/hadolint#279]
- [x] This is a bug report

### Expected behavior

* No errors when escaping variables substitution
* When ONBUILD is used, it can't pass the validation with escaping variable substitution in a string

``` Dockerfile
ONBUILD ENV RUNNER_CMD_EXEC=${RUNNER_CMD_EXEC:-"java \$JAVA_OPTS -jar /runtime/server.jar \$JAR_OPTS"}
```

According to https://docs.docker.com/engine/reference/builder/#environment-replacement, this  is valid in all instructions, plus `ONBUILD`

> Escaping is possible by adding a `\` before the variable: `\$foo` or `\${foo}`, for example, will translate to `$foo` and `${foo}` literals respectively.

### Actual behavior

```
$ hadolint runner/openjdk-10/Dockerfile
runner/openjdk-10/Dockerfile:89:54 unexpected '\' expecting '"', ''', a string with no quotes, or at least one space
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
$ hadolint --version
Haskell Dockerfile Linter 1.15.0-no-git
```

**Dockerfile** (if relevant)

```Dockerfile
ONBUILD ENV RUNNER_CMD_EXEC=${RUNNER_CMD_EXEC:-"java \$JAVA_OPTS -jar /runtime/server.jar \$JAR_OPTS"}
```

**Additional environment details (OS, stack version, etc.)**

* I've been using this format and it is working...

**Comment by lorenzo** [comment]
Fixed in v1.16.1

**Comment by lorenzo** [comment]
sorry, I closed the wrong issue

**Comment by uqcgreen** [comment]
I couldn't get around this, so I moved the var into the docker-compose.yml:
 
```
web:
    build:
      context: .
      args:
        COMPOSER_AUTH: '{"github-oauth": {"github.com": "${GITHUB_PERSONAL_ACCESS_TOKEN}"}}'
```

Then in the Dockerfile

```
ARG COMPOSER_AUTH
ENV COMPOSER_AUTH=${COMPOSER_AUTH}
```

**Comment by tdavison784** [comment]
We are currently facing this same issue with calling $JAVA_HOME in our Jenkins container build. To get around it we had to remove the "\" from the start of our $JAVA_HOME env variable call. So this works to get the docker image to build, but then breaks the linter. It would be nice to replace the escape mechanism for the special char : $ with double quotes rather than \
